File: services/api/app/common/enums.py
from enum import StrEnum


class ScheduleStatus(StrEnum):
    PENDING_DELIVERY = "pending_delivery"
    PENDING_DRAWING = "pending_drawing"
    MISSING_BOM = "missing_bom"
    PENDING_TRIGGER = "pending_trigger"
    SCHEDULABLE = "schedulable"
    SCHEDULED = "scheduled"
    SCHEDULED_STALE = "scheduled_stale"


class WarningLevel(StrEnum):
    NORMAL = "normal"
    ABNORMAL = "abnormal"


class OrderType(StrEnum):
    REGULAR = "1"
    OPTIONAL = "2"
    CUSTOM = "3"


class IssueStatus(StrEnum):
    OPEN = "open"
    RESOLVED = "resolved"
    IGNORED = "ignored"


class BomBackfillQueueStatus(StrEnum):
    PENDING = "pending"
    PROCESSING = "processing"
    RETRY_WAIT = "retry_wait"
    SUCCESS = "success"
    FAILED = "failed"
    PAUSED = "paused"


class BomBackfillFailureKind(StrEnum):
    TRANSIENT_ERROR = "transient_error"
    EMPTY_RESULT = "empty_result"
    PERMANENT_ERROR = "permanent_error"


class BackgroundTaskStatus(StrEnum):
    PENDING = "pending"
    RUNNING = "running"
    SUCCEEDED = "succeeded"
    FAILED = "failed"


class SchedulerRuntimeState(StrEnum):
    RUNNING = "running"
    PAUSED = "paused"
    STOPPED = "stopped"

File: apps/web/src/constants/enums.ts

// 排产状态
export const SCHEDULE_STATUS_MAP: Record<string, { label: string; color: string }> = {
  scheduled: { label: '已排产', color: 'success' },
  scheduled_stale: { label: '待重排', color: 'warning' },
  pending_drawing: { label: '待发图', color: 'warning' },
  pending_trigger: { label: '待触发', color: 'info' },
  schedulable: { label: '可排产', color: 'primary' },
  missing_bom: { label: '缺少BOM', color: 'danger' },
}

// 异常标识
export const WARNING_LEVEL_MAP: Record<string, { label: string; color: string }> = {
  normal: { label: '正常', color: 'success' },
  abnormal: { label: '异常', color: 'danger' },
}

// 异常状态
export const ISSUE_STATUS_MAP: Record<string, { label: string; color: string }> = {
  open: { label: '待处理', color: 'danger' },
  resolved: { label: '已解决', color: 'success' },
  ignored: { label: '已忽略', color: 'info' },
}
